src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -186,13 +176,97 @@ private void applyFix(final String polluterName,
+                return deltaDebug(polluterName, victimMethod, chunk, 2);    // If works, then delta debug some more this chunk
+            }
+            // Otherwise, check if applying complement chunk works
+            if (checkCleanerStmts(polluterName, victimMethod, otherChunk)) {
+                return deltaDebug(polluterName, victimMethod, otherChunk, n - 1);   // If works, then delta debug some more the complement chunk
+            }
+        }
+        // If not chunk/complement work, increase granularity and try again
+        return deltaDebug(polluterName, victimMethod, cleanerStmts, n * 2);
+    }
+
+    // TODO: Extract this logic out to a more generalized fixer that is separate, so we can reuse it
+    // TODO: for cleaners, santa clauses, etc.
+    private void applyFix(final String polluterName,
+                          final JavaMethod cleanerMethod,
+                          final JavaMethod victimMethod) throws Exception {
+        backup(victimMethod.javaFile());
+
+        // Compile the test without our fix
+        runMvnInstall();
+
+        // Check if we pass in isolation before fix
+        TestPluginPlugin.info("Running victim test with polluter before adding code from cleaner.");
+        final List<String> withPolluter =
+                // This is to handle the case of a santa clause, where a test will fail with no polluter
+                polluterName.isEmpty() ? ListUtil.fromArray(victimMethod.methodName()) :
+                                         ListUtil.fromArray(polluterName, victimMethod.methodName());
+        if (testOrderPasses(withPolluter)) {
+            TestPluginPlugin.error("Test passes with the polluter, but is supposed to fail.");
+            return;
+        }
+
+        // Do our fix using all cleaner code
+        TestPluginPlugin.info("Applying code from cleaner and recompiling.");
+        final NodeList<Statement> cleanerStmts = cleanerMethod.body().getStatements();
+
+        if (!checkCleanerStmts(polluterName, victimMethod, cleanerStmts)) {
+            TestPluginPlugin.error("Cleaner does not fix victim!");
+            return;
+        }
+
+        // Cleaner is good, so now we can start delta debugging
+        NodeList<Statement> minimalCleanerStmts = deltaDebug(polluterName, victimMethod, cleanerStmts, 2);
+
+        // Apply the final minimal cleaner statements
+        victimMethod.prepend(minimalCleanerStmts);
+        victimMethod.javaFile().writeAndReloadCompilationUnit();
+
+    }
+
     private boolean runMvnInstall() throws MavenInvocationException {
         // TODO: Maybe support custom command lines/options?
         final InvocationRequest request = new DefaultInvocationRequest();
         request.setGoals(Arrays.asList("install"));
         request.setPomFile(project.getFile());
         request.setProperties(new Properties());
         request.getProperties().setProperty("skipTests", "true");
+        request.getProperties().setProperty("rat.skip", "true");
```

Comment:
Actually, this addition here is not completely related to the pull request, it should have been just included into the master directly since we always want to skip random license checking like rat given that we are generating other files (backup).